File: leetcode/JavaScript实现/200-299题/0206.js
// 自己的代码
// 就是头插法，击败了100%的人。
var reverseList = function(head) {
    let dummp = new ListNode(null);
    dummp.next = head;
    let pre = dummp, p = pre.next;
    pre.next = null;
    while(p) {
        let temp = p.next;
        p.next = pre.next;
        pre.next = p;
        p = temp;
    }
    return pre.next;
};

File: leetcode/C++实现/200-299题/0206.cpp
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        if(head==NULL)
            return head;
        ListNode *p=head,*pre=NULL,*Next=NULL;
        while(p)
        {
            Next=p->next;
            p->next=pre;
            pre=p;
            p=Next;
        }
        return pre;
    }
};